src/lib/Libifl/pbsD_connect.c

```diff
@@ -932,6 +911,7 @@ pbs_connect_noblk(char *server, int tout)
 			default:
 err:
 				closesocket(sock);
+				dealloc_conn_list_single(-1);
```

Comment:
Instead of doing this, why not connect() first, and only create a new svr_connections[] if connect was successful? You won't need any of these dealloc calls then